### PR TITLE
Rename bufferSize to maxJobActive

### LINKED ISSUE #1894: Find better name for buffer size
Currently users struggle to understand what the job worker property `bufferSize` means. We should find a name which better indicates the purpose of this setting.

For migration we should keep the buffer size method and mark it as deprecated, to be removed in upcoming releases.

### PR DESCRIPTION
**Pull request by ChrisKujawa**
closes #1894

### CODE REVIEW

File: broker-core/src/main/java/io/zeebe/broker/job/JobBatchActivateProcessor.java [hunk 1]
```diff
@@ -194,11 +194,14 @@ private void rejectCommand(
...
     final String format = "Expected to activate job batch with %s to be %s, but it was %s";
 
-    if (value.getAmount() < 1) {
+    if (value.getMaxJobsToActivate() < 1) {
       rejectionType = RejectionType.INVALID_ARGUMENT;
       rejectionReason =
           String.format(
-              format, "amount", "greater than zero", String.format("'%d'", value.getAmount()));
+              format,
+              "amount",
```
Comment (menski) [inline broker-core/src/main/java/io/zeebe/broker/job/JobBatchActivateProcessor.java]:
```suggestion
              "max jobs to activate",
```

**Review by menski — CHANGES_REQUESTED** [review]
looks good, but I still found some `amount` references in docs
- https://stage.docs.zeebe.io/basics/job-workers.html#requesting-jobs-from-the-broker
- https://stage.docs.zeebe.io/reference/grpc.html#activatejobs-rpc

**Comment by Zelldon** [comment]
Yeah sorry I might missed the docs, should be fixed now

**Review by menski — APPROVED** [review]
bors r+

**Comment by menski** [comment]
bors retry

**Pull request merged**